api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java

```diff
@@ -36,8 +36,8 @@
   private static final Logger logger = Logger.getLogger(HttpTraceContext.class.getName());
 
   private static final TraceState TRACE_STATE_DEFAULT = TraceState.builder().build();
-  static final String TRACE_PARENT = "traceparent";
-  static final String TRACE_STATE = "tracestate";
+  public static final String TRACE_PARENT = "traceparent";
+  public static final String TRACE_STATE = "tracestate";
```

Comment:
if we make these public, I think we should rename them to `TRACE_PARENT_HEADER_NAME` and `TRACE_STATE_HEADER_NAME` so it's more clear what they represent.

Reply:
@mariusioana Just wondering, does the `fields` method work well for your use case without making the header names themselves public? They are all the fields for any given propagator, which may solve your case more generically?

I'm ok with making these public, with the renaming, since they're very stable constants but just wanted to check.

Reply:
@anuraaga @Oberon00 I think something like `HttpTraceContext.getInstance().fields().get(0)` does it, it's more cryptic,  but something that we can use. I would still prefer to reference like suggested by @jkwatson using constants like `TRACE_PARENT_HEADER_NAME`.

Reply:
@mariusioana why `.get(0)`? If you want to add them to the allowed-headers-list of a component, you need to add all the fields in the returned value.

Reply:
@Oberon00 true that! I was still thinking only about `Traceparent`. Btw, `Tracestate` is optional right? Reading from https://www.w3.org/TR/trace-context/#tracestate-header

Reply:
Depending on what you mean by optional. Many tracing systems will work fine without tracestate, but some might require it. Also, according to W3C, "Vendors receiving a tracestate request header MUST send it to outgoing requests.", so you may not drop it. See https://www.w3.org/TR/trace-context/#mutating-the-tracestate-field

Reply:
@mariusioana If `fields()` sounds good for you, how about we go ahead and close this PR for now? If we find another use case for these constants we can revisit, but it's easier to make private public than other way around.

Reply:
@anuraaga Sure, let's close this PR. Thanks!

Reply:
Closing it. glad to help. : )